Source/WebCore/page/Navigator.cpp

```diff
@@ -247,11 +263,29 @@ void Navigator::initializePluginAndMimeTypeArrays()
 
     auto [publiclyVisiblePlugins, additionalWebVisiblePlugins] = frame->page()->pluginData().publiclyVisiblePluginsAndAdditionalWebVisiblePlugins();
 
+    bool hasWebKitBuiltinPDFPlugin = frame->page()->pluginData().builtInPDFPlugin().has_value();
+    bool reportDummyPDFPlugin = frame->loader().client().canShowMIMEType("application/pdf"_s);
+
+    m_pdfViewerEnabled = hasWebKitBuiltinPDFPlugin || reportDummyPDFPlugin;
+
     Vector<Ref<DOMPlugin>> publiclyVisibleDOMPlugins;
     Vector<Ref<DOMPlugin>> additionalWebVisibleDOMPlugins;
     Vector<Ref<DOMMimeType>> webVisibleDOMMimeTypes;
 
-    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisiblePlugins.size());
+    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisiblePlugins.size() + dummyPDFPluginCount + (reportDummyPDFPlugin ? 1 : 0));
```

Comment:
If `!m_pdfViewerEnabled` then we should not report the dummy PDF plugin names so this doesn't seem quite right.

Reply:
Good point. Fixed.